wgpu-core/src/lib.rs

```diff
@@ -54,6 +56,56 @@ type Epoch = u32;
 
 pub type RawString = *const c_char;
 
+pub const WHOLE_SIZE: wgt::BufferAddress = !0;
+
+#[repr(transparent)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, PeekPoke)]
+#[cfg_attr(
+    feature = "trace",
+    derive(serde::Serialize),
+    serde(into = "SerBufferSize")
+)]
+#[cfg_attr(
+    feature = "replay",
+    derive(serde::Deserialize),
+    serde(from = "SerBufferSize")
+)]
+pub struct BufferSize(pub u64);
+
+impl BufferSize {
+    const WHOLE: BufferSize = BufferSize(!0u64);
+}
+
+/// This type allows us to make the serialized representation of a BufferSize more readable
+#[allow(dead_code)]
+#[cfg_attr(feature = "trace", derive(serde::Serialize))]
+#[cfg_attr(feature = "replay", derive(serde::Deserialize))]
+enum SerBufferSize {
```

Comment:
could we move this enum and its `From` implementations into `device::trace` module, where it's only used?

Reply:
👍